doc/includes/include_sidewalk_kconfig_options.txt

```diff
@@ -0,0 +1,25 @@
+* ``CONFIG_SIDEWALK`` -- Enables support for the Sidewalk protocol and its dependencies.

```

Comment:
Question about the images -> I removed the circles as they didn't have descriptions. As a good practice, the diagram shouldn't have any elements that are not explained. Preferably, I'd like to remove the white circle from "sidewalk_thread_state_machine.svg"  file, though it does have an event that leads to it, so  maybe there is some description that could be addded here?